### PR TITLE
Add node to the cluster after initial bootstrap

### PR DESCRIPTION
**Pull request by kravciak**
Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/662

Local runs:
 - http://dhcp91.suse.cz/tests/1080 - VMX
 - http://dhcp91.suse.cz/tests/1109 - DVD

### CODE REVIEW

**Comment by drpaneas** [comment]
doesn't break QAM current tests: http://skyrim.qam.suse.de/tests/overview?groupid=98&version=2.0&build=4280-kravciak&distri=caasp

If it works or not, we will test it later on. LGTM from my side and thanks for the good work as always :D

File: products/caasp/main.pm [hunk 1]
```diff
@@ -143,18 +144,22 @@ sub load_stack_tests {
...
 # Init cluster variables
 sub stack_init {
-    my $children      = get_children;
-    my $stack_size    = keys %$children;
-    my $stack_masters = $stack_size > 6 ? 3 : 1;             # For 6+ node clusters select 3 masters
-    my $stack_minions = $stack_size - $stack_masters - 1;    # Do not count admin node into minions
+    my $children       = get_children;
+    my $delayed_worker = get_delayed_worker;
+
+    my $stack_size    = (keys %$children) - !!$delayed_worker;
```
Comment (asmorodskyi) [inline products/caasp/main.pm]: is this typo or !! has some meaning in perl ?

File: tests/caasp/stack_add_nodes.pm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+sub bootstrap {
+    assert_and_click 'unassigned-select-all';
+    assert_and_click 'unassigned-add-nodes';
+    send_key 'end';
+    assert_screen 'velum-adding-nodes-done', 900;
+    send_key 'home';
+}
+
+sub run {
+    # barrier_wait {name => "DELAYED_WORKERS_INSTALLED", check_dead_job => 1};
```
Comment (asmorodskyi) [inline tests/caasp/stack_add_nodes.pm]: can you please remove this line ?

Comment (okurz) [reply tests/caasp/stack_add_nodes.pm]: +1

File: products/caasp/main.pm (hunk 1 above)
Comment (kravciak) [reply products/caasp/main.pm]: This translates variable into boolean value - https://stackoverflow.com/questions/2168406/what-does-double-exclamation-point-mean

Here it means - "if there is delayed worker, decrease $stack_size by 1"

File: tests/caasp/stack_add_nodes.pm (hunk 2 above)
Comment (kravciak) [reply tests/caasp/stack_add_nodes.pm]: removed, thanks

File: products/caasp/main.pm (hunk 1 above)
Comment (asmorodskyi) [reply products/caasp/main.pm]: thanks for clarification

**Pull request merged**